File: ui_service/frontend/src/models/stocks.ts
export interface StockDetails {
  id: string;
  symbol: string;
  name: string;
  close: number | null;
  change: number | null;
  percent_change: number | null;
  previous_close: number | null;
  open: number | null;
  high: number | null;
  low: number | null;
  volume: number | null;
  fifty_two_week_high: number | null;
  fifty_two_week_low: number | null;
  stock_summery: string | null;
}

export interface StockHistoryBar {
  date: string;
  open: number | null;
  high: number | null;
  low: number | null;
  close: number | null;
  volume: number | null;
}

export type HistoryRange = "1D" | "5D" | "1M" | "3M" | "6M" | "1Y" | "5Y";

export type ArticleSummaryStatus = "none" | "pending" | "ready" | "failed";

export interface StockArticle {
  article_id: string;
  url: string;
  title: string;
  source: string | null;
  published_at: string | null;
  provider_summary: string | null;
  ai_summary: string | null;
  ai_summary_status: ArticleSummaryStatus;
  ai_summary_error: string | null;
}

File: ui_service/backend/models/stocks.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field


class StockDetails(BaseModel):
    id: str
    symbol: str
    name: str
    close: Optional[float] = None
    change: Optional[float] = None
    percent_change: Optional[float] = None
    previous_close: Optional[float] = None
    open: Optional[float] = None
    high: Optional[float] = None
    low: Optional[float] = None
    volume: Optional[int] = None
    fifty_two_week_high: Optional[float] = None
    fifty_two_week_low: Optional[float] = None
    stock_summery: Optional[str] = None


class StockHistoryBar(BaseModel):
    date: str = Field(..., description="Trading date (YYYY-MM-DD)")
    open: Optional[float] = None
    high: Optional[float] = None
    low: Optional[float] = None
    close: Optional[float] = None
    volume: Optional[int] = None


class StockArticle(BaseModel):
    article_id: str
    url: str
    title: str
    source: Optional[str] = None
    published_at: Optional[datetime] = None
    provider_summary: Optional[str] = None
    ai_summary: Optional[str] = None
    ai_summary_status: str = Field(..., description="none | pending | ready | failed")
    ai_summary_error: Optional[str] = None
